flink-runtime/src/main/java/org/apache/flink/runtime/state/DefaultOperatorStateBackendSnapshotStrategy.java

```diff
@@ -204,7 +217,17 @@ public SnapshotResultSupplier<OperatorStateHandle> asyncSnapshot(
             if (snapshotCloseableRegistry.unregisterCloseable(localOut)) {
                 StreamStateHandle stateHandle = localOut.closeAndGetHandle();
                 if (stateHandle != null) {
```

Comment:
What if the `stateHandle == null` ?

Reply:
`SnapshotResult.of(null)` will be returned.

Theoretically, this situation cannot happen, because `registeredOperatorStatesDeepCopies` and `registeredBroadcastStatesDeepCopies` are checked at the beginning. At thisline, one of `OperatorStates` or `BroadcastStates` is definitely not empty, and there must be content written to the output.